### PR TITLE
Add CapabilityData abstraction for returned capability data

### PR DESCRIPTION
**Pull request by puiterwijk**
Signed-off-by: Patrick Uiterwijk <patrick@puiterwijk.org>

### CODE REVIEW

**Review by ionut-arm — COMMENTED** [review]
Thanks! Looks good overall, let me know what you think about the comment below

File: src/context.rs [hunk 1]
```diff
@@ -239,7 +239,7 @@ impl Context {
         capability: TPM2_CAP,
```
[the remark this answers is not in the document]
Comment (puiterwijk) [reply src/context.rs:239]: That is totally possible, and I can do so tomorrow. I wanted to get this first part of `CapabilityData` done, and was going to iterate on it.

[the remark this answers is not in the document]
Comment (ionut-arm) [reply src/context.rs:239]: Makes sense, as long as there's no immediate need for a release I'm ok with doing things "out-of-order" here, we'll just need to track these requirements somewhere. Maybe new issues, maybe one big issue with bulletpoints.

File: src/structures/capabilitydata.rs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+use crate::structures::{PcrSelect, PcrSelectionList};
+use crate::tss2_esys::*;
+use crate::{Error, Result, WrapperErrorKind};
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+
+#[derive(Debug, Clone)]
+pub enum CapabilityData {
+    Algorithms(HashMap<TPM2_ALG_ID, TPMA_ALGORITHM>),
+    Handles(Vec<TPM2_HANDLE>),
```
[the remark this answers is not in the document]
Comment (puiterwijk) [reply src/structures/capabilitydata.rs:13]: Thanks!

File: src/context.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (puiterwijk) [reply src/context.rs]: Just pushed this.

File: src/structures/capabilitydata.rs [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    ECCCurves(Vec<TPM2_ECC_CURVE>),
+    // These are in the TPM TMU_CAPABILITIES, but are not defined by esapi-2.4.1
+    // AuthPolicies(),
+    // ActData(),
+}
+
+fn cd_from_alg_properties(props: TPML_ALG_PROPERTY) -> Result<CapabilityData> {
+    let mut data = HashMap::new();
+
+    for i in 0..props.count {
```
[the remark this answers is not in the document]
Comment (puiterwijk) [reply src/structures/capabilitydata.rs:42]: Done

**Review by ionut-arm — APPROVED** [review]
Looks good to me

**Comment by Superhepper** [comment]
There are a lot of raw TPM/ESAPI types in the CapabilityData enum but I am ok with it for now but it would be good to create a ticket to see what can be removed in the future.

**Review by Superhepper — APPROVED** [review]

**Pull request merged**